chain-impl-mockchain/src/transaction/transaction.rs

```diff
@@ -112,10 +126,10 @@ impl<Extra: property::Serialize> Transaction<Address, Extra> {
         self.serialize_body(&mut codec.into_inner())
     }
 
-    pub fn hash(&self) -> TransactionId {
+    pub fn hash(&self) -> TransactionSignDataHash {
         let mut bytes = Vec::new();
         self.serialize_body(&mut bytes).unwrap();
-        TransactionId::hash_bytes(&bytes)
+        DigestOf::digest(&TransactionSignData(bytes.into()))
```

Comment:
I think there should be a better looking API, like `Digest::of` to produce `DigestOf`.
Something to think of for a future refactoring.

Reply:
ah that's a good name, will add it